File: src/main/java/kr/co/lunatalk/global/common/constants/EnvironmentConstants.java
package kr.co.lunatalk.global.common.constants;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.Arrays;
import java.util.List;

@Getter
@AllArgsConstructor
public enum EnvironmentConstants {
	PROD(Constants.PROD),
	DEV(Constants.DEV),
	LOCAL(Constants.LOCAL);

	private String value;

	@NoArgsConstructor(access = AccessLevel.PRIVATE)
	public static class Constants {
		public static final String PROD = "prod";
		public static final String DEV = "dev";
		public static final String LOCAL = "local";
		public static List<String> PROD_DEV = List.of(PROD, DEV);
	}
}

File: src/main/kotlin/kr/co/lunatalk/global/common/constants/EnvironmentConstants.kt
package kr.co.lunatalk.global.common.constants

import org.hibernate.internal.util.collections.CollectionHelper.listOf

enum class EnvironmentConstants(val value: String) {
    PROD(Constants.PROD),
    DEV(Constants.DEV),
    LOCAL(Constants.LOCAL);

    object Constants {
        const val PROD = "prod"
        const val DEV = "dev"
        const val LOCAL = "local"
        val PROD_DEV = listOf(PROD, DEV)

    }
}
